### PR TITLE
Added a advanced=bool flag to Profile() class and enabled it on the cosmic-epoch profile

### PR DESCRIPTION
**Pull request by Torxed**
This is to be able to hide certain profiles behind `--advanced`.
We also enabled it on the `cosmic-epoch` due to it being pre-release alpha still with a note on the upstream repo:

> incomplete **alpha**

This complements issue: #2358 and thus #2615

### CODE REVIEW

File: archinstall/default_profiles/profile.py [hunk 1]
```diff
@@ -138,16 +141,16 @@ def is_top_level_profile(self) -> bool:
 		return self.profile_type in top_levels
 
 	def is_desktop_profile(self) -> bool:
-		return self.profile_type == ProfileType.Desktop
+		return self.profile_type == ProfileType.Desktop if self.advanced is False or storage['arguments'].get('advanced', False) is True else False
```
Comment (svartkanin) [inline archinstall/default_profiles/profile.py:144]: nit: maybe put this evaluation in a function to make it more readable?

File: archinstall/default_profiles/desktops/cosmic.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 class CosmicProfile(XorgProfile):
 	def __init__(self):
-		super().__init__('cosmic-epoch', ProfileType.DesktopEnv, description='')
+		super().__init__('cosmic-epoch', ProfileType.DesktopEnv, description='', advanced=True)
```
Comment (svartkanin) [inline archinstall/default_profiles/desktops/cosmic.py:11]: There was also a new greeter option added in archinstall/default_profiles/profile.py does that also need to be behind the flag?

Comment (Torxed) [reply archinstall/default_profiles/desktops/cosmic.py:11]: Ah good catch, I think that should be behind the flag too. I'll update the pr with this and a fix for the other comment.

**Comment by Torxed** [comment]
Should be good to go now, tested and verified that greeter isn't visible in other DE's and the desktop profile itself isn't visible either without `--advanced`.

**Review by svartkanin — APPROVED** [review]

**Pull request merged**